pulsar-client-go/pulsar/c_consumer.go

```diff
@@ -147,7 +168,12 @@ func subscribeAsync(client *client, options ConsumerOptions, callback func(Consu
 	subName := C.CString(options.SubscriptionName)
 	defer C.free(unsafe.Pointer(subName))
 
-	callbackPtr := savePointer(&subscribeContext{conf: conf, consumer: consumer, callback: callback})
+	var callbackPtr unsafe.Pointer
+	if schema != nil {
```

Comment:
nit: I am not a go expert. but do we really need the if-else branches here? can't you just pass a `nil` schema to the `subscribeContext` struct?